runtime/wasm/src/module/mod.rs

```diff
@@ -717,9 +722,10 @@ impl WasmInstance {
         let callback: String = self.asc_get(callback);
         let user_data: store::Value = self.try_asc_get(user_data)?;
 
-        // `ipfs_map` can take a long time to process, so disable the timeout.
-        self.should_interrupt.store(false, Ordering::SeqCst);
         let flags = self.asc_get(flags);
+
+        // Pause the timeout while running ipfs_map
+        self.timeout_stopwatch.lock().unwrap().stop();
```

Comment:
This needs to use a scope-based drop API. (I've had good luck with the simplicity of the `defer` crate) As it is currently if the inner function panics or returns an error then the stopwatch is never restarted. This will result in the timeout task to be leaked.